<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>971</width>
    <height>843</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="maximumSize">
   <size>
    <width>16777215</width>
    <height>1348</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="minimumSize">
    <size>
     <width>0</width>
     <height>800</height>
    </size>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_17">
    <item>
     <widget class="QWidget" name="plot_n_log_widget" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>400</width>
        <height>700</height>
       </size>
      </property>
      <layout class="QVBoxLayout" name="plot_n_log_Vlayout">
       <item>
        <widget class="QCustomPlot" name="plot_3" native="true">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>600</width>
           <height>577</height>
          </size>
         </property>
         <property name="font">
          <font>
           <underline>true</underline>
          </font>
         </property>
         <property name="mouseTracking">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QScrollBar" name="plot_scrollbar">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QTextBrowser" name="serial_log">
         <property name="minimumSize">
          <size>
           <width>600</width>
           <height>100</height>
          </size>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <layout class="QVBoxLayout" name="sidebar_VLayout">
      <property name="sizeConstraint">
       <enum>QLayout::SetDefaultConstraint</enum>
      </property>
      <item>
       <widget class="QWidget" name="Serial_widget" native="true">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>300</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>300</width>
          <height>16777215</height>
         </size>
        </property>
        <layout class="QVBoxLayout" name="Serial_Vlayout">
         <item alignment="Qt::AlignHCenter">
          <widget class="QLabel" name="label_16">
           <property name="font">
            <font>
             <bold>true</bold>
             <underline>true</underline>
            </font>
           </property>
           <property name="text">
            <string>Serial</string>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_4">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_5">
             <item>
              <widget class="QLabel" name="label">
               <property name="text">
                <string>COM Port</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="com_port_combo"/>
             </item>
            </layout>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>Baud Rate</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="baud_combo"/>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QPushButton" name="connect_button">
           <property name="text">
            <string>Connect</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QWidget" name="MQTT_widget" native="true">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>300</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>300</width>
          <height>16777215</height>
         </size>
        </property>
        <layout class="QVBoxLayout" name="MQTT_Vlayout">
         <item>
          <widget class="QLabel" name="MQTT_Label">
           <property name="font">
            <font>
             <bold>true</bold>
             <underline>true</underline>
            </font>
           </property>
           <property name="text">
            <string>MQTT</string>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QVBoxLayout" name="mqtt_cred_Vlayout">
           <item>
            <widget class="QLabel" name="Credential_label">
             <property name="font">
              <font>
               <italic>false</italic>
               <bold>true</bold>
               <kerning>true</kerning>
              </font>
             </property>
             <property name="text">
              <string>Credentials</string>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="IP_H">
             <item>
              <widget class="QLabel" name="label_25">
               <property name="text">
                <string>IP Address</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="ip_field"/>
             </item>
             <item>
              <spacer name="horizontalSpacer_7">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item>
              <widget class="QLabel" name="label_26">
               <property name="text">
                <string>Port</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="prt_field"/>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="user_Hlayout">
             <item>
              <widget class="QLabel" name="label_3">
               <property name="text">
                <string>Username:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="username_field"/>
             </item>
             <item>
              <spacer name="horizontalSpacer">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item>
              <widget class="QLabel" name="label_4">
               <property name="text">
                <string>Client ID:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="clientID_field"/>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="pword_">
             <item alignment="Qt::AlignHCenter">
              <widget class="QLabel" name="label_5">
               <property name="text">
                <string>Password</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="password_field">
               <property name="echoMode">
                <enum>QLineEdit::Password</enum>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer_3">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QPushButton" name="mqtt_connect_button">
             <property name="text">
              <string>Connect</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QVBoxLayout" name="publish_Vlayout">
           <item>
            <widget class="QLabel" name="label_8">
             <property name="font">
              <font>
               <bold>true</bold>
              </font>
             </property>
             <property name="text">
              <string>Publish</string>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_8">
             <item>
              <widget class="QLabel" name="label_9">
               <property name="text">
                <string>Topic:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="pub_topic_field"/>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QLabel" name="label_10">
             <property name="text">
              <string>Payload</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QTextEdit" name="payload_field"/>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_7">
             <item>
              <widget class="QLabel" name="label_6">
               <property name="text">
                <string>QOS:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="qos_combo"/>
             </item>
             <item>
              <spacer name="horizontalSpacer_4">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item>
              <widget class="QLabel" name="label_7">
               <property name="text">
                <string>Retained:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="retained_combo"/>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QPushButton" name="publish_button">
             <property name="text">
              <string>Publish</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QVBoxLayout" name="mqtt_sub_Vlayout">
           <item>
            <widget class="QLabel" name="label_11">
             <property name="font">
              <font>
               <bold>true</bold>
              </font>
             </property>
             <property name="text">
              <string>Subscribe</string>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_9">
             <item>
              <widget class="QLabel" name="label_12">
               <property name="text">
                <string>Topic</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="sub_topic_field"/>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_2">
             <item>
              <widget class="QLabel" name="label_19">
               <property name="text">
                <string>QOS</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="sub_qos_combo"/>
             </item>
             <item>
              <spacer name="horizontalSpacer_2">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QPushButton" name="sub_button">
             <property name="text">
              <string>Subscribe</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QVBoxLayout" name="mqtt_unsub_Vlayout">
           <item>
            <widget class="QLabel" name="label_18">
             <property name="font">
              <font>
               <bold>true</bold>
              </font>
             </property>
             <property name="text">
              <string>Unsubscribe</string>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_11">
             <item>
              <widget class="QLabel" name="label_17">
               <property name="text">
                <string>Topic</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="unsub_topic_field"/>
             </item>
             <item>
              <widget class="QPushButton" name="unsub_button">
               <property name="text">
                <string>Unsubscribe</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QLabel" name="label_13">
             <property name="text">
              <string>Subscribed Topics</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QTextBrowser" name="subed_textb"/>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>971</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="separator"/>
    <addaction name="actionSave"/>
   </widget>
   <widget class="QMenu" name="menuSources">
    <property name="title">
     <string>Sources</string>
    </property>
    <addaction name="actionSerial"/>
    <addaction name="actionMQTT_Broker"/>
   </widget>
   <widget class="QMenu" name="menuDisplay">
    <property name="title">
     <string>Display</string>
    </property>
    <addaction name="separator"/>
    <addaction name="actionUse_Timestamp"/>
   </widget>
   <widget class="QMenu" name="menuClear">
    <property name="title">
     <string>Clear</string>
    </property>
    <addaction name="actionClear_Data"/>
   </widget>
   <widget class="QMenu" name="menuPause">
    <property name="title">
     <string>Pause</string>
    </property>
    <addaction name="actionPause_Plot"/>
    <addaction name="actionPause_Log"/>
    <addaction name="actionPause_Sampling"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuDisplay"/>
   <addaction name="menuSources"/>
   <addaction name="menuClear"/>
   <addaction name="menuPause"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionSerial">
   <property name="text">
    <string>Serial</string>
   </property>
  </action>
  <action name="actionMQTT_Broker">
   <property name="text">
    <string>MQTT Broker</string>
   </property>
  </action>
  <action name="actionUse_Timestamp">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Use Timestamp</string>
   </property>
  </action>
  <action name="actionClear_Data">
   <property name="text">
    <string>Clear Data</string>
   </property>
  </action>
  <action name="actionPause_Plot">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Pause Plot</string>
   </property>
  </action>
  <action name="actionPause_Log">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Pause Log</string>
   </property>
  </action>
  <action name="actionPause_Sampling">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Pause Sampling</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QCustomPlot</class>
   <extends>QWidget</extends>
   <header>qcustomplot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
